### PR TITLE
Update Makefile.in to create directories if they do not exist

### PR DESCRIPTION
**Pull request by abraunegg**
* Add checks to create directories if they do not exist

### CODE REVIEW

**Comment by abraunegg** [comment]
@norbusan 
This still fails on the last one - and 100% unsure why. As this you also package on Debian - is there some sort of 'trick' being missed here?
```
(dmd-2.094.1)alex@debian-10-x86-64:~/onedrive$ ./configure; make clean; make; checkinstall
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... dmd
checking version of D compiler... 2.094.1
checking for curl... yes
checking for sqlite... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating contrib/pacman/PKGBUILD
config.status: creating contrib/spec/onedrive.spec
config.status: creating onedrive.1
config.status: creating contrib/systemd/onedrive.service
config.status: creating contrib/systemd/onedrive@.service
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
if [ -f .git/HEAD ] ; then \
        git describe --tags > version ; \
else \
        echo v2.4.7-dev > version ; \
fi
dmd  -w -g -O -J. -L-lcurl -L-lsqlite3  -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d -ofonedrive

checkinstall 1.6.3, Copyright 2010 Felipe Eduardo Sanchez Diaz Duran
           This software is released under the GNU GPL.



*****************************************
**** Debian package creation selected ***
*****************************************

This package will be built according to these values: 

0 -  Maintainer: [ alex@debian-10-x86-64 ]
1 -  Summary: [ onedrive ]
2 -  Name:    [ onedrive ]
3 -  Version: [ 20201024 ]
4 -  Release: [ 1 ]
5 -  License: [ GPL ]
6 -  Group:   [ checkinstall ]
7 -  Architecture: [ amd64 ]
8 -  Source location: [ onedrive ]
9 -  Alternate source location: [  ]
10 - Requires: [  ]
11 - Recommends: [  ]
12 - Suggests: [  ]
13 - Provides: [ onedrive ]
14 - Conflicts: [  ]
15 - Replaces: [  ]

Enter a number to change any of them or press ENTER to continue: 

Installing with make install...

========================= Installation results ===========================
/usr/bin/install -c -D onedrive /usr/local/bin/onedrive
if [ ! -d "/usr/local/share/man/man1" ] ; then mkdir -p /usr/local/share/man/man1 ; fi
/usr/bin/install -c -D onedrive.1 /usr/local/share/man/man1/onedrive.1
if [ ! -d "/usr/local/etc/logrotate.d" ] ; then mkdir -p /usr/local/etc/logrotate.d ; fi
/usr/bin/install -c -D -m 644 contrib/logrotate/onedrive.logrotate /usr/local/etc/logrotate.d/onedrive
if [ ! -d "/usr/local/share/doc/onedrive" ] ; then mkdir -p /usr/local/share/doc/onedrive ; fi
mkdir: cannot create directory ‘/usr/local/share/doc’: No such file or directory
make: *** [Makefile:113: install] Error 1

****  Installation failed. Aborting package creation.

Cleaning up...OK

Bye.

(dmd-2.094.1)alex@debian-10-x86-64:~/onedrive$
```

File: Makefile.in [hunk 1]
```diff
@@ -106,9 +106,11 @@ onedrive: $(SOURCES)
 
 install: all
 	$(INSTALL) -D onedrive $(DESTDIR)$(bindir)/onedrive
+	if [ ! -d "$(DESTDIR)$(mandir)/man1" ] ; then mkdir -p $(DESTDIR)$(mandir)/man1 ; fi
 	$(INSTALL) -D onedrive.1 $(DESTDIR)$(mandir)/man1/onedrive.1
+	if [ ! -d "$(DESTDIR)$(sysconfdir)/logrotate.d" ] ; then mkdir -p $(DESTDIR)$(sysconfdir)/logrotate.d ; fi
 	$(INSTALL) -D -m 644 contrib/logrotate/onedrive.logrotate $(DESTDIR)$(sysconfdir)/logrotate.d/onedrive
-	mkdir -p $(DESTDIR)$(docdir)
+	if [ ! -d "$(DESTDIR)$(docdir)" ] ; then mkdir -p $(DESTDIR)$(docdir) ; fi
```
Comment (norbusan) [inline Makefile.in:113]: If we protect `"$(DESTDIR)..."` with quotes, then probably better to protect it in the `mkdir` call, too.

Comment (abraunegg) [reply Makefile.in:113]: Will look at that .. but what about the actual error .. any ideas?

**Comment by norbusan** [comment]
I'll look into it later

**Comment by norbusan** [comment]
Please see my discussion on the issue, this is somehow unrelated and should NOT be merged

**Comment by abraunegg** [comment]
@norbusan 
Thanks for looking into this

**Comment by norbusan** [comment]
I pushed a commit that allows installation via checkinstall, by creating all directories in advance.

**Comment by abraunegg** [comment]
@norbusan 
Looks good to me - will test under Debian 10 as per #1105

**Comment by abraunegg** [comment]
@norbusan 
Seems to have still the same issue with the doc directory ..... I have no idea why:
```
(dmd-2.094.1)alex@debian-10-x86-64:~/code/onedrive-pr1107$ ./configure; make clean; make; checkinstall
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... dmd
checking version of D compiler... 2.094.1
checking for curl... yes
checking for sqlite... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating contrib/pacman/PKGBUILD
config.status: creating contrib/spec/onedrive.spec
config.status: creating onedrive.1
config.status: creating contrib/systemd/onedrive.service
config.status: creating contrib/systemd/onedrive@.service
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
if [ -f .git/HEAD ] ; then \
        git describe --tags > version ; \
else \
        echo v2.4.7-dev > version ; \
fi
dmd  -w -g -O -J. -L-lcurl -L-lsqlite3  -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d -ofonedrive

checkinstall 1.6.3, Copyright 2010 Felipe Eduardo Sanchez Diaz Duran
           This software is released under the GNU GPL.



*****************************************
**** Debian package creation selected ***
*****************************************

This package will be built according to these values: 

0 -  Maintainer: [ alex@debian-10-x86-64 ]
1 -  Summary: [ onedrive ]
2 -  Name:    [ onedrive ]
3 -  Version: [ pr1107 ]
4 -  Release: [ 1 ]
5 -  License: [ GPL ]
6 -  Group:   [ checkinstall ]
7 -  Architecture: [ amd64 ]
8 -  Source location: [ onedrive-pr1107 ]
9 -  Alternate source location: [  ]
10 - Requires: [  ]
11 - Recommends: [  ]
12 - Suggests: [  ]
13 - Provides: [ onedrive ]
14 - Conflicts: [  ]
15 - Replaces: [  ]

Enter a number to change any of them or press ENTER to continue: 

Installing with make install...

========================= Installation results ===========================
# non GNU install or checkinstall seems to ignore the -D option
# make sure we generated the directories beforehand.
/usr/bin/install -c -d /usr/local/bin \
        /usr/local/share/man/man1 \
        /usr/local/etc/logrotate.d \
        /usr/local/share/doc/onedrive
/usr/bin/install: cannot create directory ‘/usr/local/share/doc’: No such file or directory
make: *** [Makefile:110: install] Error 1

****  Installation failed. Aborting package creation.

Cleaning up...OK

Bye.

(dmd-2.094.1)alex@debian-10-x86-64:~/code/onedrive-pr1107$ 
```

Thoughts ?

**Comment by norbusan** [comment]
That is fucking messed up, sorry. checkinstall is simply broken.

Try manual `/usr/bin/install -c -d /tmp/foo/bar/baz/bla/brumm/grummel` and all intermediate directories will be created.

If checkinstall cannot manage this, sorry, I don't consider this priority to bow to stupidity of checkinstall.

**Comment by abraunegg** [comment]
Agreed

checkinstall has many issues anyways security wise .. will close this PR and Issue Ticket

**Comment by norbusan** [comment]
Agreed!

**Pull request closed without merging**